### ISSUE TITLE
Repeated builds doing full builds every time

### ISSUE THREAD

**Issue by 68CodingMonkeys** [golang/go#47026]
#### What did you do?

1) download sources with cloning from gitbub
2) do all.bash

#### What did you expect to see?

A real build script. Meaning: 
Calling all.bash multiple times, only stuff that was not build so far is processed.
Ideally, running repeate buildes because of bugs during the build process, you
will go fastly to the problematic section.

#### What did you see instead?

All is build again. Every time.
This is important on weak devices where a build can take more than an hour.

Additional:
:
        case <-time.After(10 * time.Second): // plenty of a safety margin, even for very slow machines
:
This failed for me on raspberry 4 B. This is REALLY BAD - please remove this code.
Or make it at least optional so that a Fail does not hurt a build. Or use an estimate,
that works on all supported plattforms.

#### Does this issue reproduce with the latest release (go1.16.5)?

dont know as your build is failing from repo
raspbian seems to be only supported on paper and nobody ever has tried
to compile it on the device.

#### System details

```
go version go1.11.6 linux/arm
GOARCH="arm"
GOBIN=""
GOCACHE="/home/ethereum/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="arm"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/ethereum/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib/go-1.11"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.11/pkg/tool/linux_arm"
GCCGO="gccgo"
GOARM="6"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
GOROOT/bin/go version: go version go1.11.6 linux/arm
GOROOT/bin/go tool compile -V: compile version go1.11.6
uname -sr: Linux 5.10.17-v7l+
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
/lib/arm-linux-gnueabihf/libc.so.6: GNU C Library (Debian GLIBC 2.28-10+rpi1) stable release version 2.28.
gdb --version: GNU gdb (Raspbian 8.2.1-2) 8.2.1
```

**Comment by thanm** [comment]
Hello,

I think there may be some misunderstanding here about the role of the various scripts in $GOROOT/src.

Running **all.bash** will do a build of your distribution and also run all the tests; testing typically takes much longer than the build.

If you just want to build Go, run "**make.bash**".

Once **make.bash** is complete, you can then do an incremental build via

`   go install std cmd
`

That command will only rebuild as necessary (e.g. works incrementally).

Hope this helps.

**Comment by bcmills** [comment]
> Additional:
> :
> case <-time.After(10 * time.Second): // plenty of a safety margin, even for very slow machines

Please file a separate issue for that. (I've been pushing on eliminating arbitrary timeouts throughout the Go project.)

**Comment by 68CodingMonkeys** [comment]
> Hello,
> 
> I think there may be some misunderstanding here about the role of the various scripts in $GOROOT/src.
> 
> Running **all.bash** will do a build of your distribution and also run all the tests; testing typically takes much longer than the build.
> 
> If you just want to build Go, run "**make.bash**".
> 
> Once **make.bash** is complete, you can then do an incremental build via
> 
> ` go install std cmd`
> 
> That command will only rebuild as necessary (e.g. works incrementally).
> 
> Hope this helps.

Oh man, I wished I knew that before since a single build took me 93 minutes.
Patience you will learn.

**Comment by mvdan** [comment]
Seems like this is resolved.
